sound/soc/sof/intel/tgl.c

```diff
@@ -106,6 +106,8 @@ int sof_tgl_ops_init(struct snd_sof_dev *sdev)
 		sof_tgl_ops.dbg_dump	= hda_ipc4_dsp_dump;
 
 		sof_tgl_ops.set_power_state = hda_dsp_set_power_state_ipc4;
+
+		sof_tgl_ops.get_stream_position = hda_dsp_get_stream_hda_link_position;
```

Comment:
I don't know how this can work.

the stream position is always given by DPIB. 
The link position is irrelevant in case of mixing, etc.
We have to rely on the firmware information to know WHEN the stream was added into a mixer and deal with the delay report from the firmware.

Reply:
The `get_stream_position` is to get the _link_ position, introduced by
2cd67feff14f ("ASoC: SOF: add get_stream_position ops for pcm delay")
and the MTL implementation is
fe43dab744c0 ("ASoC: SOF: add get_stream_position support on mtl platforms")

LNL inherited this.

Reply:
With this patch CAV2.5 and AVS1 behaves in a same way. If the LLP information is not available from firmware then we use the HDA registers.

Reply:
yes we behave in the same way, but how good is it in the first place? You still need the information provided by the firmware to get an offset. Just try with the deep buffer output enabled 10s later than the regular output, this could be fun.

Reply:
The start offset is recorded, I have tested it.